layer.go

```diff
@@ -74,10 +74,28 @@ const (
 	CycloneDXJSON SBOMFormat = iota
 	SPDXJSON
 	SyftJSON
+	UnknownFormat
 )
 
 func (b SBOMFormat) String() string {
-	return []string{"cdx.json", "spdx.json", "syft.json"}[b]
+	return []string{"cdx.json", "spdx.json", "syft.json", "unknown"}[b]
+}
+
+func (b SBOMFormat) MimeType() string {
+	return []string{"application/vnd.cyclonedx+json", "application/spdx+json", "application/vnd.syft+json", "unknown"}[b]
+}
+
+func SBOMFormatFromString(from string) (SBOMFormat, error) {
```

Comment:
Should we create a slice of all known sbom formats? That way we can just iterate over them.

This will also help with the validate sbom function where we can check the file suffix to match with the known formats. Lmk what you think.

Reply:
Sounds reasonable. Go question for you about this. How would you achieve this? I thought you couldn't have a const slice. I know I could define the strings as const and use them in the slice, maybe that's better? Is there a common pattern for doing something like this that you've seen? Thanks

Reply:
I moved the strings to consts & used the consts in the array. I did a quick google search and const arrays/slices are not a thing, as far as I can tell, so I'm not sure we can do much more. I don't think using a global var would be good. I think that could potentially allow someone to modify the list, which wouldn't be desirable, IMHO.

Reply:
https://github.com/dmarkham/enumer looks like a good way to manage auto generation of all these methods.

Reply:
That project looks interesting, but I don't see a way to do custom transformations. If we were going from SomeConst and wanted the string representation to be `SomeConst` or `some_const` or `some-const`, it looks like we could use this. From what I see in the docs, there's no custom transformation to make `SomeConst` turn into `some.json` and vise versa, which is what we need.

It [also looks like it's generating a global var](https://github.com/dmarkham/enumer/blob/master/enumer.go#L104) which would have the same limitation, someone could potentially mess with the list.

The lists are fairly small and shouldn't grow a ton over time. It seems like we could keep them hand-written and not take on too much burden, IMHO (adding dependencies for this functionality also has its own risks). 

I'll acknowledge that there is some room for error with using the arrays to convert from enum to string, and a new type needs to be added. If you forget to update the arrays, you'll get a runtime panic. I could change that to use switch statements instead. It's more code, but when a type is added if you forget to update the conversion methods, you will end up with an unknown value instead of a runtime panic so it's a little safer. Would that be preferred?